service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/ClonePolicyAttributesStep.java

```diff
@@ -79,9 +81,26 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
 
   @Override
   public StepResult undoStep(FlightContext context) throws InterruptedException {
-    // var destinationAttributes =
-    //  context.getWorkingMap().get(WorkspaceFlightMapKeys.POLICIES, ApiTpsPolicyInputs.class);
-    // TODO: PF-2019 - when we have a replace method, replace the destination PAO attributes
+    BearerToken token = new BearerToken(userRequest.getRequiredToken());
+    var destinationAttributes =
```

Comment:
If the `do` step fails before writing to the `POLICIES` key I think this will cause problems - it looks like this will call replacePao with a null `newAttributes` in the request, which I think will just wipe all policies.

Reply:
whoops, missed the latest commit that already fixed this!